lib/icinga/checkable-dependency.cpp

```diff
@@ -78,10 +73,14 @@ bool Checkable::IsReachable(DependencyType dt, Dependency::Ptr *failedDependency
 	std::unordered_map<std::string, Dependency::Ptr> violated; // key: redundancy group, value: nullptr if satisfied, violating dependency otherwise
 
 	for (const Dependency::Ptr& dep : deps) {
-		std::string redundancy_group = dep->GetRedundancyGroup();
+		bool parentReachable{true};
+		if (Checkable::Ptr parent = dep->GetParent(); parent.get() != this) {
+			parentReachable = parent->IsReachable(dt, failedDependency, rstack + 1);
```

Comment:
Ah, because currently the sum of all direct and intermediate dependencies is limited to `256`. You want me to change this so that the limit applies to each dependency object individually? I can do this, but as you already said, it's not related to this issue.